src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,83 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import {
+  FC,
+  Dispatch,
+  SetStateAction,
+  ChangeEvent,
+} from 'react';
+import { TodoStatus } from '../../types/TodoStatus';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string,
+  setQuery: Dispatch<SetStateAction<string>>,
+  setTodoStatus: Dispatch<SetStateAction<TodoStatus>>,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: FC<Props> = ({
+  query,
+  setQuery,
+  setTodoStatus,
+}) => {
+  const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    switch (e.target.value) {
+      case 'all':
+        setTodoStatus(TodoStatus.ALL);
+        break;
+      case 'active':
+        setTodoStatus(TodoStatus.ACTIVE);
+        break;
+      case 'completed':
+        setTodoStatus(TodoStatus.COMPLETED);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={handleChange}
+          >
+            {Object.values(TodoStatus).map((enumValue) => (
+              <option
+                key={enumValue}
+                value={enumValue}
+              >
+                {enumValue}
```

Comment:
in example value of filters have first uppercased letter, idk if it really matters, but let it be:)

Reply:
Tests are expected values to be in lowercase. I could possibly add a name using slice method but i think it`s not so critical in this case